### ISSUE TITLE
Can't replicate results

### ISSUE THREAD

**Issue by jambobjambo** [xiechenxi99/MaSS13K#2]
I attempted to recreate the results you shared in your paper, but I struggled to achieve the same results.

Firstly, I downloaded the dataset as you mentioned in the repo.
Used the training script provided.
https://github.com/xiechenxi99/MaSS13K/blob/main/mmsegmentation/configs/massformer/massformer_r50_8xb2-90k_mass13k-1024x1024.py

I'm training on one GPU rather than four, and my batch size is 2, so I'm curious if that's the problem.

After training for 80k iterations, my testing class had these results;
+------------+-------+-------+-------+
|   Class    |  IoU  |  Acc  |  BIoU |
+------------+-------+-------+-------+
| background |  16.7 |  36.8 |  3.47 |
|   person   | 22.11 | 37.32 |  2.47 |
|  building  | 18.91 | 27.59 |  4.27 |
|    tree    | 55.22 |  71.1 | 17.83 |
|   ground   | 47.53 | 72.92 |  5.77 |
|    sky     |  0.72 |  0.73 |  0.29 |
|   water    |  0.0  |  0.0  |  0.0  |
+------------+-------+-------+-------+

My loss during training was hovering around 50.

I noticed in your training script you use a learning rate of 0.0003, but in your paper you state that you use an initial learning rate of 0.003.
Therefore, I made this change to account for the fact that I am using 1 GPU instead of 4 and added accumulative_counts to 16 in the optim_wrapper to simulate a batch size of 32.

Yet after 60k iterations I am still getting pretty bad results.

+------------+-------+-------+------+
|   Class    |  IoU  |  Acc  | BIoU |
+------------+-------+-------+------+
| background |  4.11 |  7.82 | 0.81 |
|   person   |  0.0  |  0.0  | 0.0  |
|  building  | 25.69 | 96.11 | 5.65 |
|    tree    |  0.0  |  0.0  | 0.0  |
|   ground   |  0.0  |  0.0  | 0.0  |
|    sky     |  0.0  |  0.0  | 0.0  |
|   water    |  0.0  |  0.0  | 0.0  |
+------------+-------+-------+------+

Am I doing something very wrong? Is there a way of training on one GPU, or should I switch to a 4-GPU system, and it will work?
Could you please share your pre-trained weights so I can verify them against yours?

**Comment by xiechenxi99** [comment]
Thank you for your attention to our work. We used a batch size of 16, and both 4x4 GPUs or 2x8 GPUs are applicable. We are sorry that we haven't tried using a smaller batch size with gradient accumulation to simulate large-batch training, so we're unsure about the performance of single-GPU training. Additionally, if gradient accumulation is used to increase the effective batch size, we suppose the number of iterations should be correspondingly increased. In addition, we appreciate your pointing out the typo in the learning rate in the paper—we will correct the typo in the paper, and the learning rate provided in the training script **3e-4** is the correct setting.  

https://drive.google.com/file/d/1kDgoHf3XBDhE4LQ_KhJnSiZN9etb2cbh/view?usp=share_link
We have provided the checkpoint from the latest training based on the given code (with minimal differences compared to the best results) to assist you in identifying potential bugs. Perhaps you could visualize the code execution to check if it runs correctly. Please feel free to contact us if you encounter any further issues.
